server/modules/elastic/elasticassistantstore.go

```diff
@@ -0,0 +1,533 @@
+// Copyright 2020-2025 Security Onion Solutions LLC and/or licensed to Security Onion Solutions LLC under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0 as shown at
+// https://securityonion.net/license; you may not use this file except in compliance with the
+// Elastic License 2.0.
+
+package elastic
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/apex/log"
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/security-onion-solutions/securityonion-soc/model"
+	"github.com/security-onion-solutions/securityonion-soc/server"
+	modcontext "github.com/security-onion-solutions/securityonion-soc/server/modules/context"
+	"github.com/security-onion-solutions/securityonion-soc/util"
+	"github.com/security-onion-solutions/securityonion-soc/web"
+)
+
+type ElasticAssistantstore struct {
+	server       *server.Server
+	esClient     *elasticsearch.Client
+	chatIndex    string
+	sessionIndex string
+	schemaPrefix string
+	maxLogLength int
+}
+
+func NewElasticAssistantstore(srv *server.Server, client *elasticsearch.Client, maxLogLength int) *ElasticAssistantstore {
+	return &ElasticAssistantstore{
+		server:       srv,
+		esClient:     client,
+		maxLogLength: maxLogLength,
+	}
+}
+
+func (store *ElasticAssistantstore) Init(chatIndex string, sessionIndex string, schemaPrefix string) error {
+	store.chatIndex = chatIndex
+	store.sessionIndex = sessionIndex
+	store.schemaPrefix = schemaPrefix
+
+	return nil
+}
+
+func (store *ElasticAssistantstore) save(ctx context.Context, obj any, index string, kind string) (*model.EventIndexResults, error) {
+	document := ConvertObjectToDocumentMap(kind, obj, store.schemaPrefix)
+	document[store.schemaPrefix+"kind"] = kind
+
+	results, err := store.indexDoc(ctx, index, document)
+
+	return results, err
+}
+
+func (store *ElasticAssistantstore) indexDoc(ctx context.Context, index string, document map[string]any) (*model.EventIndexResults, error) {
+	logger := log.FromContext(ctx)
+
+	results := model.NewEventIndexResults()
+
+	request, err := convertToElasticIndexRequest(document)
+	if err == nil {
+		var response string
+
+		logger.Debug("Sending index request to primary Elasticsearch client")
+		response, err = store.indexDocument(ctx, store.disableCrossClusterIndex(index), request)
+		if err == nil {
+			err = convertFromElasticIndexResults(response, results)
+			if err != nil {
+				logger.WithError(err).Error("Encountered error while converting document index results")
+			}
+		} else {
+			logger.WithError(err).Error("Encountered error while indexing document into elasticsearch")
+		}
+	}
+
+	return results, err
+}
+
+func (store *ElasticAssistantstore) indexDocument(ctx context.Context, index string, document string) (string, error) {
+	logger := log.FromContext(ctx)
+
+	logger.WithFields(log.Fields{
+		"documentIndex": index,
+		"document":      store.truncate(document),
```

Comment:
This log could be a problem since it appears to be logging the chat message. I might be misunderstanding what the document var represents though.

Reply:
I know we have some other debug statements that also log a truncated elastic response as well so perhaps we need to hide these behind a dev mode toggle.